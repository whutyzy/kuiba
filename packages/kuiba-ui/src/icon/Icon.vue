<template>
    <i @click="$emit('click', $event)" :class="iconClass" :style="iconStyle"></i>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue'
import { addUnit } from '../utils/format/unit'

export default defineComponent({
    name: 'KuibaIcon',
    props: {
        name: { type: String },
        color: { type: String },
        size: { type: [String, Number] },
        classPrefix: { type: String, default: 'kuiba-icon' }
    },
    emits: ['click'],
    setup(props) {
        const iconClass = computed(() => `${props.classPrefix} ${props.classPrefix}-${props.name}`)
        const iconStyle = computed(() => ({ color: props.color, fontSize: addUnit(props.size) }))
        return { iconClass, iconStyle }
    }
})
</script>

<style lang="scss">
@import './icon.scss';
</style>
