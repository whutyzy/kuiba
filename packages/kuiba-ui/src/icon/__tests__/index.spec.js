import example from '../example'
import Icon from '../index'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'

test('test icon example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test icon plugin', () => {
  const app = createApp({}).use(Icon)
  expect(app.component(Icon.name)).toBeTruthy()
})

test('test icon click', () => {
    
  const clickFn = jest.fn()
  const wrapper = mount({
    template: `<kuiba-icon @click="clickFn"></kuiba-icon>`,
    components: { [Icon.name]: Icon },
    methods: {
      clickFn,
    },
  })
  wrapper.trigger('click')
  expect(clickFn).toHaveBeenCalledTimes(1)
  wrapper.unmount()
})

test('test icon name', () => {
  const wrapper = mount(Icon, { props: { name: 'arrow-left' } })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test icon namespace', () => {
  const wrapper = mount(Icon, { props: { prefixClass:'iconfont',name: 'arrow-left' } })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test icon style', () => {
  const wrapper = mount(Icon, { props: { name: 'arrow-left', color: 'red', size: '12px' } })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})
