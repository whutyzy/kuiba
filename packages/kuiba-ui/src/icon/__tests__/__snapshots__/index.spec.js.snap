// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test icon example 1`] = `
"<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">基础用法</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"icon-row\\">
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-left\\"></i></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-right\\"></i></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-up\\"></i></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-down\\"></i></div>
    </div>
  </div>
</div>
<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">图标颜色</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"icon-row\\">
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-left\\"></i></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-left\\" style=\\"color: orange;\\"></i></div>
    </div>
  </div>
</div>
<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">图标大小</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"icon-row\\">
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-left\\" style=\\"font-size: 16px;\\"></i></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-left\\" style=\\"font-size: 32px;\\"></i></div>
    </div>
  </div>
</div>
<div class=\\"kuiba-demo-block\\">
  <h4 class=\\"kuiba-demo-block__title\\">图标列表</h4>
  <div class=\\"kuiba-demo-block__container\\">
    <div class=\\"icon-row\\">
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-right\\"></i><span>arrow-right</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-left\\"></i><span>arrow-left</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-up\\"></i><span>arrow-up</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-arrow-down\\"></i><span>arrow-down</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-success\\"></i><span>success</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-cross\\"></i><span>cross</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-plus\\"></i><span>plus</span></div>
      <div class=\\"icon-example-box\\"><i class=\\"kuiba-icon kuiba-icon-warning\\"></i><span>warning</span></div>
    </div>
  </div>
</div>"
`;

exports[`test icon name 1`] = `"<i class=\\"kuiba-icon kuiba-icon-arrow-left\\"></i>"`;

exports[`test icon namespace 1`] = `"<i class=\\"kuiba-icon kuiba-icon-arrow-left\\" prefixclass=\\"iconfont\\"></i>"`;

exports[`test icon style 1`] = `"<i class=\\"kuiba-icon kuiba-icon-arrow-left\\" style=\\"color: red; font-size: 12px;\\"></i>"`;
